### PR TITLE
Add payload parameter support for onMessage method

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/2880
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java [hunk 1]
```diff
@@ -272,19 +298,84 @@ public void testInvalidService15() {
...
+            assertDiagnostic(diagnostic, INVALID_FUNCTION_PARAM_PAYLOAD);
+        }
+    }
+
+    @Test
+    public void testInvalidService20() {
+        Package currentPackage = loadPackage("invalid_service_20");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.errors().size(), 2);
```
Comment (Bhashinee) [inline compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java:361]:
```suggestion
        Assert.assertEquals(diagnosticResult.errorCount(), 2);
```

File: compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java [hunk 2]
```diff
@@ -272,19 +298,84 @@ public void testInvalidService15() {
...
+        assertDiagnostic(diagnostic, INVALID_FUNCTION);
     }
 
     @Test
     public void testInvalidService16() {
         Package currentPackage = loadPackage("invalid_service_16");
         PackageCompilation compilation = currentPackage.getCompilation();
         DiagnosticResult diagnosticResult = compilation.diagnosticResult();
-        Assert.assertEquals(diagnosticResult.errors().size(), 3);
+        Assert.assertEquals(diagnosticResult.errors().size(), 2);
```
Comment (Bhashinee) [inline compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java:309]:
```suggestion
        Assert.assertEquals(diagnosticResult.errorCount(), 2);
```

Comment (Bhashinee) [reply compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java:309]: Let's use this in other places as well.

Comment (dilanSachi) [reply compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java]: Updated in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/393/commits/b981317fec0692ca8130a099798591997bf0bba1

File: compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java (hunk 1 above)
Comment (dilanSachi) [reply compiler-plugin-tests/src/test/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqCompilerPluginTest.java]: Updated in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/393/commits/b981317fec0692ca8130a099798591997bf0bba1

**Comment by Bhashinee** [comment]
LGTM.

**Review by Bhashinee — APPROVED** [review]

**Pull request merged**